common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -460,9 +460,12 @@ def running(request):
 def parse_query_params(strategy, response, *args, **kwargs):
     """Reads whitelisted query params, transforms them into pipeline args."""
     auth_entry = strategy.request.session.get(AUTH_ENTRY_KEY)
-    if not (auth_entry and auth_entry in _AUTH_ENTRY_CHOICES):
-        raise AuthEntryError(strategy.request.backend, 'auth_entry missing or invalid')
-
+    if not auth_entry:
+        # If auth_entry is not in the session, we got here by a non-standard workflow.
+        # We simply assume 'login' in that case.
+        return {'auth_entry': AUTH_ENTRY_LOGIN}
+    if auth_entry not in _AUTH_ENTRY_CHOICES:
+        raise AuthEntryError(strategy.request.backend, 'auth_entry invalid')
```

Comment:
```
# If auth_entry is not in the session, we got here by a non-standard workflow.
# We simply assume 'login' in that case.
auth_entry = strategy.request.session.get(AUTH_ENTRY_KEY, AUTH_ENTRY_LOGIN)
if auth_entry not in _AUTH_ENTRY_CHOICES:
    raise AuthEntryError(strategy.request.backend, 'auth_entry invalid')
```

Reply:
That's not quite the same, since the original code handled the case that `auth_entry` is the empty string differently than what you propose.  However, thinking about it, it seems more reasonable to treat `auth_entry being the empty string as an error, so I'm changing the code.